### ISSUE TITLE
Choice of type of filter by column ? (non automatic filter, slider by default)

### ISSUE THREAD

**Issue by philibe** [rstudio/DT#569]
Hello,

__My cases__

I would like to choose the type of filter by column :)

- I would like to not have automatic slider for numeric. My users don't like sliders because they don't search by range but by value, except some cases. So it's counter intuitive to us to type `2.3 ... 2.3`, they would like `2.3` for example.
- Same thing for date. I convert nearly all dates to string to workaround that feature to disable date slider.
- I don't understand why string column are filtered like numeric if the datas of the current column is without character. I prefix with` ':'` to force non automatic numeric filter which is not a good thing for ids.
- and ... Is there a way to have automatic listbox by column ? :)

**TLDR :)** 
I am aware that all I suggest take time. I would like only to not have automatic filter if it is too complicated or too long.

My workaround to not have automatic filter, for the column I want,  is to cast the column to string, but I cannot do that for numeric.

Have a good day :p 

__My wishes :)  except if it does exist but I didn't see it in rstudio/DT or datables.net__

My wishes are to kept filter features but to disable automatic filter by type of variable, and to not have slider by default. Have I misread documentation or it doesn't exist ?  :)

For example in my display definition, to kept  `search`, `filter`:
```
  output$my_table <- DT::renderDataTable(
    DT::datatable( my_datas,
                   style = "bootstrap",   class = "compact", filter='top',
                   selection = c("single"),    
                   options = list(
                     deferRender = TRUE, 
                     bSortClasses = TRUE,iDisplayLength = 10,   width = "100%",
                     scrollX=TRUE,
                     autoWidth = TRUE,
                     lengthMenu  = list(c(5, 25, 50, 75, 100, -1), list('5', '25','50','75','100', 'All')),
                     search = list(
                       smart = TRUE,
                       regex = TRUE, 
                       caseInsensitive = TRUE
                     )
                   )
    )   
  )
```

but to have something like that for the filter
```
search = list
(
type = no_auto
)
```
where type would be `auto`, `no_auto` (without slider,  like string filter)

and by column to have something like that for the filter
```
options = list(searchCols = list(
    list(type =string ),  # even if it's numeric
    list(type = auto),  # current DT shiny feature
    list(type = no_auto),
    list(type = range),  # current DT shiny feature for date and numeric
    list(type = no_range),   # for dates or numeric
     list(type = listbox_auto),   # automatic list box
     list(type = listbox list (a,b,c,d)),   # automatic list box
  ))
```

**Comment by philibe** [comment]
> My users don't like sliders because they don't search by range but by value, except some cases. So it's counter intuitive for them to type 2.3 ... 2.3, they would like 2.3 for example.
>Same thing for date.

Workaround for search number without interval for a unique number value. ( And for dates also).

```javascript
function search_replace_interval(e) {
    if(e.keyCode == 13) {
        before=$(this).val();
        after=before;
        // prefix by the $ if you are looking for a number without interval : 
        //$34 will be replaced by 34...34
        if (before.substring(0,1)=='$') {
          item=before.substring(1,before.length);
          after=item+'...'+item; 
      }
      $(this).val(after)
    }
}

$(".datatables thead [data-type~='number'] input.form-control").on("keydown",search_replace_interval);
$(".datatables thead [data-type~='date'] input.form-control").on("keydown",search_replace_interval);
```
